.github/workflows/main.yml

```diff
@@ -54,17 +60,56 @@ jobs:
         env:
           ENABLE_SEARCH: 'true'
 
+      # TODO fails when run before build but would be better before (issues with the generated markdown component overrides)
       - name: Lint text files
         run: yarn run jest --projects jest.{eslint,stylelint,test,text}.config.js --maxWorkers=3 --reporters jest-silent-reporter
 
+      - name: Initialize Deployment
+        uses: bobheadxi/deployments@v0.6.2
+        id: start-deployment
+        with:
+          step: start
+          token: ${{ secrets.GITHUB_TOKEN }}
+          env: ${{ env.DEPLOY_ENV }}
+          ref: ${{ github.head_ref }}
+
+      - name: Deploy to Vercel
+        id: deploy-to-vercel
+        uses: amondnet/vercel-action@v20
+        with:
+          vercel-project-name: commercetools-docs-kit
+          vercel-token: ${{ secrets.VERCEL_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          vercel-args: ${{ env.VERCEL_ARGS }}
+          vercel-org-id: ${{ secrets.VERCEL_ORG_ID}}
+          scope: commercetools
+          vercel-project-id: ${{ secrets.VERCEL_PROJECT_ID}}
+          # there are length restrictions to domains and only *.commercetools.vercel.app works
+          # reliably for generated subdomains. No branch name domain because branch names can be long.
+          alias-domains: |
+            docskit-sha-${{ github.sha }}.commercetools.vercel.app
+            docskit-pr-{{PR_NUMBER}}.commercetools.vercel.app
```

Comment:
This might be an issue when running a production build on main.  I guess the action will resolve it to `docskit-pr-.commercetools.vercel.app`  which is not a real issue but might collide with later production deploys

Reply:
I would suggest to split the "Deploy to Vercel" step into 2:

* one that is configured for production (no alias domains, passing the `--prod` flag)
* one that is configured for previews

It's a bit more verbose but it's more explicit which config is used for what. Then we also don't need to do things like `echo "VERCEL_ARGS=--prod" >> $GITHUB_ENV`

Reply:
PS: will un-do this again because getting the deployment URL from two different steps conditionally is just as messy